### PR TITLE
Misc size limit fix

### PR DESCRIPTION
**Pull request by yfguo**
## Pull Request Description

This is mainly related to how we define the `MPIDI_OFI/POSIX_MAX_AM_HDR_SZ`. This size controls how much am_hdr buffer is allocated inline in the am_request_header_t of OFI and POSIX. The current value is smaller than what is representable by the am_hdr_sz bits (8 bits total) in the OFI/POSIX am msg header. So when the am_hdr_sz is larger than this macro, we will malloc a buffer and use it for am_hdr rather than using what is inline of the request header.

Since the request header is allocated from a pool of 1KB cells. I don't think it makes much sense to keep the inline am_hdr buffer to be small. So this PR just change it to the maximum representable size.

Another change is to account for the possible padding size when calculating eager limit for POSIX. 

## Author Checklist
* [ ] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [ ] Remove xfail from the test suite when fixing a test
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Passes whitespace checkers
* [x] Passes warning tests
* [ ] Passes all tests
* [x] Add comments such that someone without knowledge of the code could understand
* [ ] You or your company has a signed contributor's agreement on file with Argonne
* [ ] For non-Argonne authors, request an explicit comment from your companies PR approval manager

### CODE REVIEW

**Comment by yfguo** [comment]
test:mpich/ch4/most

**Review by raffenet — APPROVED** [review]
LGTM. Glad to see those calls to `malloc` go away.

**Comment by yfguo** [comment]
test:mpich/ch4/most

retest after rebase, will merge once it is cleared.

**Pull request merged**
